File: backend/hackathon-be/src/main/java/com/likelion/hackatonbe/domain/dailylog/dto/DailyLogDto.java
package com.likelion.hackatonbe.domain.dailylog.dto;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

public class DailyLogDto {

    @Getter
    @Setter
    @NoArgsConstructor
    public static class CreateRequest {
        private String mealType;     // BREAKFAST, LUNCH, DINNER, SNACK
        private List<String> foods;
        private LocalDate date;
    }

    @Getter
    public static class Response {
        private Long id;
        private String mealType;
        private List<String> foods;
        private String imageUrl;
        private LocalDate date;

        public Response(Long id, String mealType, List<String> foods, String imageUrl, LocalDate date) {
            this.id = id;
            this.mealType = mealType;
            this.foods = foods;
            this.imageUrl = imageUrl;
            this.date = date;
        }
    }
}

File: mobile/AtoCueMobile/app/src/main/java/com/example/atocuemobile/network/dto/DailyLogDto.kt
package com.example.atocuemobile.network.dto

// 서버에 일상/식단 기록을 보낼 때 사용하는 요청 데이터
data class DailyLogCreateRequest(
    val mealType: String,       // BREAKFAST, LUNCH, DINNER, SNACK
    val foods: List<String>,
    val showerCount: Int,
    val moisturizerCount: Int,
    val symptoms: List<String>,
    val memo: String,
    val date: String            // "YYYY-MM-DD"
)

// 서버에서 응답으로 받는 데이터
data class DailyLogResponse(
    val id: Long,
    val mealType: String,
    val foods: List<String>,
    val imageUrl: String?,
    val showerCount: Int,
    val moisturizerCount: Int,
    val symptoms: List<String>,
    val memo: String,
    val date: String
)